akka-sample-sharding-scala/src/main/scala/sample/sharding/TemperatureService.scala

```diff
@@ -0,0 +1,65 @@
+package sample.sharding
+
+import scala.concurrent.duration._
+import scala.util.Random
+
+import akka.actor.typed.{ ActorRef, Behavior }
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.typed.{ ClusterShardingSettings, ShardingEnvelope }
+import akka.cluster.sharding.typed.scaladsl.{ ClusterSharding, Entity, EntityTypeKey }
+import akka.cluster.typed.{ Cluster, Join }
+
+object TemperatureService {
+
+  // Update a random device
+  case object UpdateDevice extends Message
+  case object ReadTemperatures extends Message
+
+  val TypeKey: EntityTypeKey[Message] =
+    EntityTypeKey[Message]("Device")
+
+  def apply(): Behavior[Message] =
+    Behaviors.setup { context =>
+      val cluster = Cluster(context.system)
+      cluster.manager ! Join(cluster.selfMember.address)
+
+      val numberOfDevices = ClusterShardingSettings(context.system).numberOfShards
```

Comment:
That would mean/expect one device per shard, which isn't a typical usage of sharding. If you don't want to complicate it with an additional config property you can use a hardcoded value here. Just an example.

Reply:
I thought that's what the sample was doing before - maybe I was wrong. Updating.